src/main/java/uk/co/ramp/statistics/StatisticsRecorderImpl.java

```diff
@@ -0,0 +1,83 @@
+package uk.co.ramp.statistics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import uk.co.ramp.io.types.StandardProperties;
+import uk.co.ramp.people.Case;
+import uk.co.ramp.statistics.types.ImmutableInfection;
+import uk.co.ramp.statistics.types.ImmutableRValueOutput;
+import uk.co.ramp.statistics.types.Infection;
+
+public class StatisticsRecorderImpl implements StatisticsRecorder {
+
+  private final Map<Integer, Integer> personDaysIsolation = new HashMap<>();
+  private final Map<Integer, Integer> peopleInfected = new HashMap<>();
+  private final Map<Integer, Integer> contactsTraced = new HashMap<>();
+  private final Map<Integer, List<Infection>> r0Progression = new HashMap<>();
```

Comment:
I would suggest passing in the empty maps through constructor. This would improve the test readability as you would be able to pass in a populate hashmap and then call get rolling average (as an example)